core/server/api/canary/identity.js

```diff
@@ -0,0 +1,33 @@
+const settings = require('../../services/settings/cache');
+const urlUtils = require('../../lib/url-utils');
+const jwt = require('jsonwebtoken');
+const jose = require('node-jose');
+const issuer = urlUtils.urlFor('admin', true);
+
+const dangerousPrivateKey = settings.get('ghost_private_key');
+const keyStore = jose.JWK.createKeyStore();
+const keyStoreReady = keyStore.add(dangerousPrivateKey, 'pem');
+
+const getKeyID = async () => {
+    const key = await keyStoreReady;
+    return key.kid;
+};
+
+const sign = async (claims, options) => {
+    const kid = await getKeyID();
+    return jwt.sign(claims, dangerousPrivateKey, Object.assign({
+        issuer,
+        expiresIn: '30m',
+        audience: '@TODO@',
+        algorithm: 'RS256',
+        keyid: kid
+    }, options));
+};
+
+module.exports = {
+    async read(frame) {
+        // @TODO Must check the role of the user - e.g. Owner, Administrator...
```

Comment:
@allouis  What is the usecase for this? If it's just the owners that should be able to access this endpoint then don't see the reason to include in the signature. 
Also, for simplicity, was thinking of throwing our standard 403 in case this wasn't an owner user making the request.

Reply:
We want the consumer of the token to be able to determine if the token was created by Owner. So either we include the role or only give to Owner.

I think only owner to access like you said is gut :)

Reply:
Cool. Removed the comment as only allowing access for the owner for now :+1: